internal/preflight/preflight.go

```diff
@@ -0,0 +1,99 @@
+package preflight
+
+import (
+	"context"
+	"fmt"
+
+	kmmv1beta1 "github.com/qbarrand/oot-operator/api/v1beta1"
+	"github.com/qbarrand/oot-operator/internal/auth"
+	"github.com/qbarrand/oot-operator/internal/module"
+	"github.com/qbarrand/oot-operator/internal/registry"
+
+	"k8s.io/apimachinery/pkg/types"
+	ctrlruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	VerificationStatusReasonBuildConfigPresent = "Verification successful, all driver-containers have paired BuildConfigs in the recipe"
+	VerificationStatusReasonNoDaemonSet        = "Verification successful, no driver-container present in the recipe"
+	VerificationStatusReasonUnknown            = "Verification has not started yet"
+	VerificationStatusReasonVerified           = "Verification successful, this Module would be verified again in this Preflight CR"
+)
+
+//go:generate mockgen -source=preflight.go -package=preflight -destination=mock_preflight_api.go
+
+type PreflightAPI interface {
+	PreflightUpgradeCheck(ctx context.Context, mod *kmmv1beta1.Module, kernelVersion string) (bool, string)
+}
+
+func NewPreflightAPI(
+	client client.Client,
+	registryAPI registry.Registry,
+	kernelAPI module.KernelMapper) PreflightAPI {
+	return &preflight{
+		registryAPI: registryAPI,
+		kernelAPI:   kernelAPI,
+		client:      client,
+	}
+}
+
+type preflight struct {
+	client      client.Client
+	registryAPI registry.Registry
+	kernelAPI   module.KernelMapper
+}
+
+func (p *preflight) PreflightUpgradeCheck(ctx context.Context, mod *kmmv1beta1.Module, kernelVersion string) (bool, string) {
```

Comment:
Could this just return an error?

Reply:
same comment as above. error wont' tell us much, we need the status and the reason message for the status